### ISSUE TITLE
User lock batch job

### ISSUE THREAD

**Issue by Programmer-yar** [turian/audio-discrimination-crowdsource#30]
Depends on #29 
The annotations are read and users that are
performing too poorly are locked

**Comment by turian** [comment]
This is not at all ready.

**Comment by Programmer-yar** [comment]
I actually didn't quite get this spec, can you please elaborate.
Like how will you read annotations and compare them?
Do you want to get list of all annotations using API and read them wherever you want and then if you want to lock the user you want to make another API call to the user endpoint and lock that user?

**Comment by turian** [comment]
Something like this. Please flesh it out as a more formal spec (API names, parameters, preconditions, postconditions, and return values).

There's one admin-only method where they can basically get a dump of the entire database. But through a JSON API, not directly hitting postgres. (Yeah I know this sounds "expensive", but TBH the DB will probably be under 50k given this is small scale so don't worry about speed. We can optimize it later.)

There's another method where the admin can lock users with particular ids.

**Comment by turian** [comment]
TBH I think you should split this into two stories. The first is just a method where the admin can retrieve the entire database as JSON. Maybe something similar to `./manage.py dumpdata > db.json` but without writing to disk.

I want an API method to get the database and do offline batch processing on my laptop.

Then story two is just list a bunch of user IDs to freeze.

**Comment by Programmer-yar** [comment]
Ok
We will need 2 api endpoints for this
One will get the list of all annotations using GET method
other will use PUT mehtod to edit existing user status
I will soon share the details of these 2 endpoints

**Comment by Programmer-yar** [comment]
Authentication method `TokenAuthentication`
and Permission `IsAuthenticated` and `IsAdminUser` are global for all API endpoints from PR containing this issue

Details about annotation API endpoint name `annotation-api`.
- Method `GET`
- endpoint `polls/api/v1/annotation-list/`
- successful request returns status code of `200` and list of annotations
- In returned JSON, each JSON object represents one row of annotation in the table
- As 'task` and `user` are ForeignKey relations they contain `id` of `task` or `user` entry to which they are linked, this representation can be customized.
![Screenshot from 2022-09-19 14-44-09](https://user-images.githubusercontent.com/62334099/190991922-d4a53814-ed28-4665-be4f-ffe462aecc84.png)

**Comment by Programmer-yar** [comment]
Lock users API endpoint
- Name `lock-users-api`.
- Method `POST`
- endpoint `/api/v1/lock-users/`
- lock users in posted list
- requires json payload containing `id's` of users to lock
```
json_payload = {
    "users": [1, 4, 10]
    }
```
- successful request returns status code of `200` and list of user ids not found in db, if any
![lock_users](https://user-images.githubusercontent.com/62334099/190994824-21437b6a-5fc7-4417-8c1e-bb5d60c18087.png)

**Comment by DeFidelity** [comment]
This is completed and merged now.
